network/l2.c

```diff
@@ -196,56 +191,83 @@ static inline void network_add_egress_port(uint8_t *payload, uint8_t port) {
   @param *tx_evt tx event with buffer, port, and other information
 */
 static void bm_l2_process_tx_evt(L2QueueElement *tx_evt) {
-  if (tx_evt) {
-    uint8_t *payload = (uint8_t *)bm_l2_get_payload(tx_evt->buf);
-    for (uint8_t port = 0; port < CTX.num_ports; port++) {
-      if (tx_evt->port_mask & (0x01 << port)) {
-        uint8_t bm_egress_port = 0x01 << port;
-        network_add_egress_port(payload, bm_egress_port);
+  if (tx_evt == NULL) {
+    return;
+  }
+
+  uint8_t *payload = (uint8_t *)bm_l2_get_payload(tx_evt->buf);
+  const uint8_t *dst_ip = &payload[ipv6_destination_address_offset];
+  if (is_global_multicast(dst_ip)) {
+    const uint8_t all_ports = 0;
```

Comment:
Actually I now see a problem that I need to fix. When we receive a global multicast packet, we need to forward it to other ports but not retransmit it on the received port. Here I'm assuming global packets always go to all ports, which isn't true.